avogadro/qtplugins/selectiontool/selectiontool.cpp

```diff
@@ -226,30 +225,76 @@ void SelectionTool::applyColor(Vector3ub color)
   m_molecule->emitChanged(Molecule::Atoms);
 }
 
+void SelectionTool::selectLinkedMolecule(QMouseEvent* e, Index atom)
+{
+  std::queue<Index> toSelect;
+  std::set<Index> done;
+  toSelect.push(atom);
+  while (!toSelect.empty()) {
+    atom = toSelect.front();
+    toSelect.pop();
+    selectAtom(e, atom);
+    auto bonds = m_molecule->bonds(atom);
```

Comment:
This is probably a useful iterator "get connected component." I'm working on some iterators over the angles and torsions, so I'll look at adding a range iterator for "visit all atoms connected to this" and some similar things.

Reply:
Note this is `O(atoms*bonds*log(atoms))` and for large structures may be slow. maybe a dinamic solution using a structure ID and pointers should be considerated.